### DISCUSSION TITLE
Force ClickHouse to execute all subquries on local tables while distributed query

### DISCUSSION THREAD

**Discussion by alschastny** [ClickHouse/ClickHouse#34363] [Q&A] [answered]
Hello everyone!

How can I force ClickHouse to execute a query on a distributed table so that all subqueries are executed on local tables on the same shards? 

Example:
`SELECT some_simple_aggregate_function(field1) FROM distributed_table WHERE some_function(some_field, (SELECT some_state FROM local_table WHERE ...)) GROUP BY some_key`

Each distributed_table shard has its own local_table.
I know about the LOCAL / GLOBAL IN / JOIN mechanism - it does not helps here.

**Comment by UnamedRus** [comment, marked as answer]
cluster function + view() function

https://clickhouse.com/docs/en/sql-reference/table-functions/view/#view

**Reply by alschastny** [reply]
Nice hack! Thank you.
